packages/eas-cli/src/build/ios/version.ts

```diff
@@ -97,6 +99,15 @@ export async function bumpVersionInAppJsonAsync({
   }
 }
 
+function validateShortVersion(shortVersion: string | undefined): string | undefined {
+  if (shortVersion && !SHORT_VERSION_REGEX.test(shortVersion)) {
+    throw new Error(
+      `CFBundleShortVersionString must be a period-separated list of three non-negative integers. Current value: ${shortVersion}`
```

Comment:
A lot of managed projects users will not know what that value means, maybe add a different message for managed projects that refers to app.json field directtlly or e.g.
"CFBundleShortVersionString (version field in app.json/app.config.js) must be a period-separated list of three non-negative integers."